source/src/molecular-unfolding/cdk/construct-batch-evaluation.ts

```diff
@@ -81,6 +82,47 @@ export class BatchEvaluation extends Construct {
       enableKeyRotation: true,
     });
 
+    const snsKey = new kms.Key(this, 'SNSKey', {
+      enableKeyRotation: true,
+    });
+
+    snsKey.addToResourcePolicy(new iam.PolicyStatement({
+      actions: [
+        'kms:Decrypt',
+        'kms:GenerateDataKey',
+      ],
+      principals: [new iam.ServicePrincipal('events.amazonaws.com')],
+      resources: ['*'],
```

Comment:
why using wildcard resource?

Use [high level method](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kms.Key.html#grantwbrencryptgrantee) to follow best practice

Reply:
it must be set to '*', doc https://aws.amazon.com/premiumsupport/knowledge-center/sns-not-getting-eventbridge-notification/